index.js

```diff
@@ -27,16 +29,42 @@ const main = (fn, fp, data, opts) => {
 		});
 	}
 
-	const json = JSON.stringify(data, opts.replacer, opts.indent);
+	return fn(fp, data, opts);
+};
+
+const readFile = fp => {
+	return pathExists(fp)
+		.then(exists => exists ? pify(fs.readFile)(fp, 'utf8') : null);
+};
+
+const main = (fp, data, opts) => {
+	return (opts.auto ? readFile(fp) : Promise.resolve())
```

Comment:
Could also do the following if we wanted to get rid of that redundant `Promise.resolve()`:

```js
const writeFile = str => {
    const indent = str ? detectIndent(str).indent : opts.indent;
    const json = JSON.stringify(data, opts.replacer, indent);

    return pify(writeFileAtomic)(fp, `${json}\n`, {mode: opts.mode});
};

...

return opts.auto ? readFile(fp).then(writeFile) : writeFile();
```